src/components/views/DownloadModModal.vue

```diff
@@ -103,24 +103,36 @@ import ProfileModList from '../../r2mm/mods/ProfileModList';
                             }
                         );
                     } catch (e) {
+                        store.commit('download/updateDownload', { assignId, failed: true });
                         return reject(e);
                     }
                     await ProfileModList.requestLock(async () => {
+                        let currentDownloadIndex = 0;
                         for (const combo of downloadedMods) {
                             try {
                                 await DownloadModModal.installModAfterDownload(profile, combo.getMod(), combo.getVersion());
                             } catch (e) {
+                                store.commit('download/updateDownload', { assignId, failed: true });
+                                currentDownloadIndex++;
```

Comment:
I think this does nothing as we `return` on the next line? A bit confusing for the reader though, so consider removing if it's useless.